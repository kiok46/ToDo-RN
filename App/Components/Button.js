import React, { Component } from 'react';
import { Text, View, TouchableOpacity, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import Colors from '../Utils/Colors';


class Button extends Component {

	render (){
		return (
		  <TouchableOpacity onPress={this.props.onPress}>
            <View style={styles.containerStyle}>
              <Text style={styles.textStyle}>
                    { this.props.text }
              </Text>
            </View>
          </TouchableOpacity>
		);
	}
};

const styles = StyleSheet.create({
    containerStyle: {
        backgroundColor: Colors.tintColor,
        paddingHorizontal: 10,
        paddingVertical: 10,
        borderRadius: 5,
        justifyContent: 'center',
        alignItems: 'center',
        margin: 5
      },
    textStyle: {
        fontSize: 18,
        color: Colors.white,
    }
});


Button.defaultProps = {
    text: "",
    onPress: () => {}
}

Button.propTypes = {
    text: PropTypes.string,
    onPress: PropTypes.func
}

export default Button;