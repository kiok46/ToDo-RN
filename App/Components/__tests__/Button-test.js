import React from 'react';
import Button from '../Button';
import { shallow } from 'enzyme';


describe('testing Button rendering', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <Button />
        )
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ text: "Hello" })
        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ onPress: () => {} })
        expect(wrapper).toMatchSnapshot();
    })
})
