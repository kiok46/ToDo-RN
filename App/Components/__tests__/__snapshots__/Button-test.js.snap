// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button rendering renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onPress={[Function]}
    text=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2f95dc",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            }
          }
        >
          
        </Text>
      </View>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            }
          }
        >
          
        </Text>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#2f95dc",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 5,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 18,
          },
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2f95dc",
              "borderRadius": 5,
              "justifyContent": "center",
              "margin": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
              }
            }
          >
            
          </Text>
        </View>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
              }
            }
          >
            
          </Text>,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#2f95dc",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            },
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`testing Button rendering renders as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onPress={[Function]}
    text="Hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2f95dc",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            }
          }
        >
          Hello
        </Text>
      </View>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            }
          }
        >
          Hello
        </Text>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#2f95dc",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 5,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Hello",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 18,
          },
        },
        "ref": null,
        "rendered": "Hello",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2f95dc",
              "borderRadius": 5,
              "justifyContent": "center",
              "margin": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
              }
            }
          >
            Hello
          </Text>
        </View>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
              }
            }
          >
            Hello
          </Text>,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#2f95dc",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Hello",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            },
          },
          "ref": null,
          "rendered": "Hello",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`testing Button rendering renders as expected 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onPress={[Function]}
    text="Hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2f95dc",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            }
          }
        >
          Hello
        </Text>
      </View>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            }
          }
        >
          Hello
        </Text>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#2f95dc",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 5,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Hello",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 18,
          },
        },
        "ref": null,
        "rendered": "Hello",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2f95dc",
              "borderRadius": 5,
              "justifyContent": "center",
              "margin": 5,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
              }
            }
          >
            Hello
          </Text>
        </View>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
              }
            }
          >
            Hello
          </Text>,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#2f95dc",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 5,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Hello",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 18,
            },
          },
          "ref": null,
          "rendered": "Hello",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
